### ISSUE TITLE
cannot unmarshal DNS message (TLS SRV) 

### ISSUE THREAD

**Issue by 21andrewmitchell** [etcd-io/etcd#12814]
**Summary**

I have observed an issue using DNS SRV to bootstrap the etcd cluster

```
couldn't resolve during SRV discovery (error querying DNS SRV records for _etcd-server-ssl lookup xxxxxxxxxx on 127.0.0.53:53: cannot unmarshal DNS message)

error setting up initial cluster: error querying DNS SRV records for _etcd-server-ssl lookup xxxxxxxxx on 127.0.0.53:53: cannot unmarshal DNS message
```

Initially seen this issue with version 3.4.15 but seems to have been present for sometime as I can only get it work with 3.3.13.

**Expected Behaviour**

ETCD cluster should be initialised without the above error

**Actual Behaviour**

Cluster fails to bootstrap using SRV due to the issue mentioned above

This seems to have been reported before without being addressed : https://github.com/etcd-io/etcd/issues/11376

**Comment by brenix** [comment]
We encountered this a while back. If I recall correctly, there had been a bug in systemd-resolved that returned a different format for SRV records, which results in the error from the OP. It appears this should have been resolved since systemd v240+.

Since we run etcd under a container, the workaround we used was to pass the `--dns` flag to the container via docker (with our DNS server) to bypass systemd-resolved completely.

Hope that helps any future visits to this issue!
